super_editor/example/lib/demos/example_editor/example_editor.dart

```diff
@@ -145,19 +158,97 @@ class _ExampleEditorState extends State<ExampleEditor> {
     // I tried explicitly unfocus()'ing the URL textfield
     // in the toolbar but it didn't return focus to the
     // editor. I'm not sure why.
-    _editorFocusNode!.requestFocus();
+    _editorFocusNode.requestFocus();
+  }
+
+  DocumentGestureMode get _gestureMode {
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.android:
+        return DocumentGestureMode.android;
+      case TargetPlatform.iOS:
+        return DocumentGestureMode.iOS;
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.linux:
+      case TargetPlatform.macOS:
+      case TargetPlatform.windows:
+        return DocumentGestureMode.mouse;
+    }
+  }
+
+  bool get _isMobile => _gestureMode != DocumentGestureMode.mouse;
+
+  DocumentInputSource get _inputSource {
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.android:
+      case TargetPlatform.iOS:
+        return DocumentInputSource.ime;
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.linux:
+      case TargetPlatform.macOS:
+      case TargetPlatform.windows:
+        return DocumentInputSource.keyboard;
+    }
   }
 
+  void _cut() => _docOps.cut();
+  void _copy() => _docOps.copy();
+  void _paste() => _docOps.paste();
+  void _selectAll() => _docOps.selectAll();
+
   @override
   Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Expanded(
+          child: _buildEditor(),
+        ),
+        if (_isMobile) _buildMountedToolbar(),
+      ],
+    );
+  }
+
+  Widget _buildEditor() {
     return SuperEditor(
-      editor: _docEditor!,
+      editor: _docEditor,
       composer: _composer,
       focusNode: _editorFocusNode,
       scrollController: _scrollController,
       documentLayoutKey: _docLayoutKey,
       maxWidth: 600, // arbitrary choice for maximum width
       padding: const EdgeInsets.symmetric(vertical: 56, horizontal: 24),
+      gestureMode: _gestureMode,
+      inputSource: _inputSource,
+      androidToolbarBuilder: (_) => AndroidTextEditingFloatingToolbar(
+        onCutPressed: _cut,
+        onCopyPressed: _copy,
+        onPastePressed: _paste,
+        onSelectAllPressed: _selectAll,
+      ),
+      iOSToolbarBuilder: (_) => IOSTextEditingFloatingToolbar(
+        onCutPressed: _cut,
+        onCopyPressed: _copy,
+        onPastePressed: _paste,
+      ),
+    );
+  }
+
+  Widget _buildMountedToolbar() {
+    final selection = _composer.selection;
+
+    if (selection == null) {
+      return const SizedBox();
+    }
+
+    return MultiListenableBuilder(
+      listenables: <Listenable>{
+        _doc,
+        _composer.selectionNotifier,
+      },
+      builder: (_) => KeyboardEditingToolbar(
+        document: _doc,
+        composer: _composer,
+        commonOps: _docOps,
+      ),
```

Comment:
With the current code, `KeyboardEditingToolbar` will never be visible on mobile, because we're not listening the selection and rebuilding appropriately.

Something like this fixes it (but surfaces a bug where the `KeyboardEditingToolbar` is visible even though the keyboard isn't):

```suggestion
    return MultiListenableBuilder(
      listenables: <Listenable>{
        _doc,
        _composer.selectionNotifier,
      },
      builder: (_) { 
        final selection = _composer.selection;

        if (selection == null) {
          return const SizedBox();
        }

        return KeyboardEditingToolbar(
          document: _doc,
          composer: _composer,
          commonOps: _docOps,
        );
      },
```

Reply:
Ok, I was able to reproduce and I made the suggested change. Thanks.